app/services/hashtag_autocomplete_service.rb

```diff
@@ -9,43 +9,55 @@ def self.search_conditions
   end
 
   attr_reader :guardian
-  cattr_reader :data_sources, :contexts
 
-  def self.register_data_source(type, klass)
-    @@data_sources[type] = klass
+  def self.data_sources
+    # Category and Tag data sources are in core and always should be
+    # included for searches and lookups.
+    [CategoryHashtagDataSource, TagHashtagDataSource] |
```

Comment:
We could potentially lift the default `data_sources` and `contextual_type_priorities` up into class constants. What do you think about making them sets as well? Then we have some guarantees of no duplicates without requiring developers to need to use `#|` and/or `#uniq`.

Reply:
Yep this makes sense!